lib/httpRequestBuilder.js

```diff
@@ -43,6 +43,11 @@ function requestBuilder (defaults) {
       host = hostname + ':' + port
     }
 
+    if (reqData.auth) {
+      const encodedAuth = Buffer.from(reqData.auth).toString('base64')
```

Comment:
I think it would be better to parse these from the URL instead, e.g. `http://matteo:collina@localhost:3000` will generate `Buffer.from(matteo:collina).toString('base64')`

Reply:
I'm not quite sure what you mean, afaik the URL is parsed here https://github.com/mcollina/autocannon/blob/master/lib/run.js#L79 and passed to the `Client` constructor [here](https://github.com/mcollina/autocannon/blob/master/lib/run.js#L215) so it's already parsed from the URL :)

Reply:
Oh, would you mind adding a full test that verifies it all works as expectedv